AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/ViewNotifierHelper.java

```diff
@@ -49,11 +58,25 @@ public JVar replacePreviousNotifier(JBlock block) {
 		AbstractJClass notifierClass = holder.getEnvironment().getJClass(OnViewChangedNotifier.class);
 		if (notifier == null) {
 			notifier = holder.getGeneratedClass().field(PRIVATE | FINAL, notifierClass, "onViewChangedNotifier" + generationSuffix(), _new(notifierClass));
-			holder.getGeneratedClass()._implements(HasViews.class);
+			implementHasViewsInHolder();
 		}
 		return block.decl(notifierClass, "previousNotifier", notifierClass.staticInvoke("replaceNotifier").arg(notifier));
 	}
 
+	private void implementHasViewsInHolder() {
+		holder.getGeneratedClass()._implements(HasViews.class);
+		JCodeModel codeModel = environment.getCodeModel();
+
+		JDirectClass genericType = codeModel.directClass("T");
+		JMethod findViewById = holder.getGeneratedClass().method(PUBLIC, genericType, "internalFindViewById");
+		findViewById.generify("T", environment.getClasses().VIEW);
+		findViewById.annotate(Override.class);
+
+		JVar idParam = findViewById.param(codeModel.INT, "id");
+		IJExpression findViewByIdExpression = holder.getFindViewByIdExpression(idParam);
+		findViewById.body()._return(JExpr.cast(genericType, findViewByIdExpression));
```

Comment:
Additional Note: It looks like the AppCompatActivity does not yet have the generic `findViewById` so the cast would be needed on android O + appcompat activity.

i checked the code in [android-o-preview-2](https://android.googlesource.com/platform/frameworks/support/+/android-o-preview-2/v7/appcompat/src/android/support/v7/app/AppCompatActivity.java#189) and [master](https://android.googlesource.com/platform/frameworks/support/+/master/v7/appcompat/src/android/support/v7/app/AppCompatActivity.java#189) branches.